server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1633,6 +1640,223 @@ public PrimaryDataStoreInfo cancelPrimaryStorageForMaintenance(CancelPrimaryStor
         return (PrimaryDataStoreInfo)_dataStoreMgr.getDataStore(primaryStorage.getId(), DataStoreRole.Primary);
     }
 
+    @Override
+    @ActionEvent(eventType = EventTypes.EVENT_SYNC_STORAGE_POOL, eventDescription = "synchronising storage pool with management server", async = true)
+    public StoragePool syncStoragePool(SyncStoragePoolCmd cmd) {
+        Long poolId = cmd.getPoolId();
+        StoragePoolVO pool = _storagePoolDao.findById(poolId);
+
+        if (pool == null) {
+            String msg = "Unable to obtain lock on the storage pool record while syncing storage pool with management server";
```

Comment:
Suggestion:

We could improve logging here by adding the `poolId` in the message.

Reply:
added pool id.